### PR TITLE
refactor!: turn `Badge` to standalone and move in kit package

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the new behaviour?

update `Badge` migration
turn `Badge` to standalone directive
move `Badge` to `kit` package
move `Appearance` directive, `Icons` directive, `IconResolver` to `core` package

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/kit/components/badge/badge.style.less [hunk 1]
```diff
@@ -1,321 +1,120 @@
...
-
-    --left-content-color: currentColor;
-    position: relative;
-    display: inline-flex;
-    font: var(--tui-font-text-s);
-    align-items: center;
-    justify-content: center;
-    box-sizing: border-box;
+// TODO: Change to [tuiBadge] in 4.0
+[tuiBadgeNew] {
```
Comment (nsbarsukov) [inline projects/kit/components/badge/badge.style.less:4]: Will it be done in separate PR ?

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/components/badge/badge.style.less (hunk 1 above)
Comment (vladimirpotekhin) [reply projects/kit/components/badge/badge.style.less]: fixed 👍
